bundles/framework/timeseries/view/TimeSeriesRangeControlPlugin.js

```diff
@@ -68,6 +71,49 @@ class TimeSeriesRangeControlPlugin extends BasicMapModulePlugin {
     _createControlElement () {
         return jQuery('<div class="mapplugin timeseriesrangecontrolplugin"></div>');
     }
+
+    _createEventHandlers () {
+        return {
+            AfterMapMoveEvent: (event) => this.updateDataYears()
+        };
+    }
+
+    updateDataYears () {
+        const sandbox = this.getSandbox();
+        const layer = this._delegate.getLayer();
+        const options = layer.getOptions();
+        const timeseries = options.timeseries || {};
+        const metadata = timeseries.metadata || {};
+        const layerId = metadata.layer;
+        if (!layerId) {
+            return;
+        }
+
+        const url = Oskari.urls.getRoute('GetWFSFeatures', {
+            id: layerId,
+            bbox: sandbox.getMap().getBboxAsString(),
+            srs: sandbox.getMap().getSrsName()
+        });
+        fetch(url, {
+            method: 'GET',
+            headers: {
+                'Accept': 'application/json'
+            }
+        }).then(response => {
+            if (!response.ok) {
+                this._loc('messages.errorFetchTimeSeriesMetadata');
+            }
+            return response.json();
+        }).then(json => {
+            const yearSet = new Set();
+            const { features } = json;
+            features.forEach(feature => {
+                const year = moment(feature.properties.time).year();
```

Comment:
The attribute name to use for `time` can be configured by the admin as part of the metadata. `time` can be used as a default value but you should check the metadata what the admin has configured for the attribute name.